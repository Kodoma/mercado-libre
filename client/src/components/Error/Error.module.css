.errorContainer {
    margin: 0 auto;
    position: relative;
    display: flex;
    flex-direction: column;
    align-content: center;
    align-items: center;
    text-align: center;
}
