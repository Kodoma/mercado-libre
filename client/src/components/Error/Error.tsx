import React from 'react';
import styles from './Error.module.css';
import { Typography } from '@material-ui/core';
import NotFound from '../../../public/images/svg/notfound.svg';

const ErrorComponent = () => {
    return (
        <div className={styles.errorContainer}>
            <NotFound />
            <Typography variant={'h6'}>Parece que esta p√°gina no existe</Typography>
        </div>
    );
};

export default React.memo(ErrorComponent);
