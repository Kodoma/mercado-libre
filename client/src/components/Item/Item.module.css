.root {
    margin: 0 10%;
    display: flex;
    flex-direction: column;
    min-width: fit-content;
    width: 500px;
    background-color: #EEEEEE;
}

.cover {
    cursor: zoom-in;
    height: 480px;
    object-fit: contain;
    width: 380px;
}
.details {
    display: flex;
    flex-direction: column;
}
.content {
    flex: 1 0 auto;
}

.customTag {
    max-width: 100%;
    height: 500px;
    background: black;
}

.imageGallery {
    width: 25%;
}

.data {
    padding-left: 10%;
    padding-top: 5%;
    color: #333333;
}
.sold {
    color: #707070;
}

@media (max-width: 812px) {
    .root {
        width: 100%;
        min-width: auto;
    }
    .data {
        padding: 0px 0px 15% 0px;
        margin-left: auto !important;
        margin-right: auto !important;
    }
}
