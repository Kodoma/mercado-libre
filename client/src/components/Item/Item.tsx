import React from 'react';
import { Item } from '../../models/items';
import fetch from 'cross-fetch';
import styles from './Item.module.css';
import { Button, Card, CardContent, Grid, Typography } from '@material-ui/core';
import Breadcrumb from '../BreadCrumb/Breadcrumb';
import ImageGallery from './ImageGallery/ImageGallery';
import Price from '../Items/Price/Price';
import { isMobile } from 'react-device-detect';
import { useRouter } from 'next/router';
import Loading from '../Loading/Loading';

const ItemProduct = () => {
    const [item, setItem] = React.useState(ItemProductInitState);
    const router = useRouter();
    const { id } = router && router.query;

    React.useEffect(() => {
        async function fetchItemProduct(id: string) {
            if (!id) return;
            const request = await fetch('http://0.0.0.0:8080/api/items/' + id);
            if (request.status === 200) {
                const response = await request.json();
                setItem(response.item);
            } else {
                await router.push('/error');
            }
        }

        fetchItemProduct(id as string);
    }, [id]);

    const content = (
        <div className={styles.root}>
            <Breadcrumb search={item && item.title} />
            <Card>
                <CardContent>
                    <Grid
                        container
                        alignItems={isMobile ? 'center' : 'flex-start'}
                        direction={isMobile ? 'column' : 'row'}>
                        <Grid item>
                            <ImageGallery pictures={item.pictures} />
                        </Grid>
                        <Grid item xs={isMobile ? 1 : 6} className={styles.data}>
                            <Grid item xs>
                                <Grid item xs>
                                    <Typography variant="subtitle1" className={styles.sold}>
                                        {`${item.condition} - ${item.sold_quantity} vendidos`}
                                    </Typography>
                                    <Typography variant="h6" gutterBottom>
                                        {item.title}
                                    </Typography>
                                    <Price
                                        id={item.id}
                                        amount={item.price.amount}
                                        currency={item.price.currency}
                                        free_shipping={item.free_shipping}
                                    />
                                    <Button
                                        variant="contained"
                                        color="primary"
                                        fullWidth={isMobile ? true : false}>
                                        Comprar
                                    </Button>
                                </Grid>
                            </Grid>
                        </Grid>
                        {item.description.length > 0 && (
                            <Grid container direction="row">
                                <Typography variant="h6" gutterBottom>
                                    Descripcion del producto.
                                </Typography>
                                <Typography variant="body2" color="textSecondary">
                                    {item.description}
                                </Typography>
                            </Grid>
                        )}
                    </Grid>
                </CardContent>
            </Card>
        </div>
    );

    return <React.Fragment>{item === ItemProductInitState ? <Loading /> : content}</React.Fragment>;
};

export default React.memo(ItemProduct);

const ItemProductInitState: Item = {
    id: '',
    title: '',
    price: {
        amount: '',
        currency: ''
    },
    thumbnail: '',
    pictures: [],
    condition: '',
    free_shipping: false,
    sold_quantity: 0,
    description: ''
};
