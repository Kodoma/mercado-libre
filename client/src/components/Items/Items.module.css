.paper {
    padding: 5%;
    margin: 0 auto;
    display: flex;
    min-width: fit-content;
    border: none;
    boxShadow: none;
}

.custom {
    border: none;
    boxShadow: none;
}

.image {
    outline: none;
    width: 160px;
}

.img {
    object-fit: contain;
    display: block;
    aspect-ratio: auto 160 / 160;
}

.priceContainer {
    display: flex;
    align-content: center;
    align-items: center;
}

.freeShipping {
    margin-left: 10px;
}

.price,
.price:hover,
.price:link,
.price:visited,
.price:active,
.price:hover {
    color: #333333;
    text-decoration: none !important;
    outline: none !important;
}

.title,
.title:hover,
.title:link,
.title:visited,
.title:active,
.title:hover {
    color: #333333 !important;
    text-decoration: none !important;
    outline: none !important;
}

.condition {
    color: #878787
}

@media (max-width: 812px) {
    .paper {
        width: 100%;
        min-width: auto;
    }

}
