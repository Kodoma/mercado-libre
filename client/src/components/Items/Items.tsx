import React from "react";
import fetch from "cross-fetch";
import { useRouter } from 'next/router'
import Link from 'next/link'
import styles from './Items.module.css';
import {Box, Card, CardContent, CircularProgress, Grid, Paper, Typography} from "@material-ui/core";
import {Item} from "../../models/items";
import Price from "./Price/Price";
import Title from "./Title/Title";
import Breadcrumb from "../BreadCrumb/Breadcrumb";
import {isMobile} from "react-device-detect";
import Loading from "../Loading/Loading";


const ItemsList = () => {
    const [items, setItems] = React.useState(ItemsInitState);
    const router = useRouter();
    const search = (router) && router.query.search

    React.useEffect(() => {
        async function fetchItems(search: string) {
            if(!search) return;
            const request = await fetch(
                "http://0.0.0.0:8080/api/items?search="+search
            );
            const response = await request.json();
            setItems(response.items);
        }

        fetchItems(search as string);
    },[search]);

    const content = items && items.map( i => (
        <Card
            className={styles.custom}
            square={true}
            key={i.id}
        >
            <CardContent>
                <Paper
                    elevation={0}
                    className={styles.paper}>
                    <Grid container
                          spacing={2}
                          justify="center"
                          alignContent="center"
                          alignItems={(isMobile) ? "center": "flex-start"}
                          direction={(isMobile) ? "column": "row"}
                    >
                        <Grid item>
                            <Link
                                href={{
                                    pathname: `/items/${i.id}`,
                                }}
                            >
                                <img width={160} height={160} className={styles.img} src={i.thumbnail}  alt={i.title}/>
                            </Link>
                        </Grid>
                        <Grid item xs={12} sm container>
                            <Grid item xs container direction="column" spacing={2}>
                                <Grid item xs>
                                    <Price id={i.id} amount={i.price.amount} currency={i.price.currency} free_shipping={i.free_shipping}/>
                                    <Title id={i.id} title={i.title} />
                                </Grid>
                            </Grid>
                            <Grid item>
                                <Typography variant="subtitle1" className={styles.condition}>
                                    { i.condition }
                                </Typography>
                            </Grid>
                        </Grid>
                    </Grid>
                </Paper>
            </CardContent>
        </Card>
    ))

    return(
        <Box width={(isMobile) ? "100%" : "75%"}>
            <Breadcrumb search={search as string} />
            { (content.length === 0) ? <Loading /> : content }
        </Box>
    );
}

export default ItemsList

const ItemsInitState: Item[] = [];
