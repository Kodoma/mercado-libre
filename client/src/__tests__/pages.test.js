import React from 'react';
import Index from '../../pages/index';
import Items from '../../pages/items/index';
import Item from '../../pages/items/[id]';
import { cleanup } from '@testing-library/react';
import { shallow } from 'enzyme';
import Error from "../../pages/error";

describe('Snapshot Pages', () => {
    afterEach(cleanup);

    it('renders homepage unchanged', () => {
        const component = shallow(<Index />);
        expect(component.getElements()).toMatchSnapshot();
    });

    it('renders items list page unchanged on route items/search=iphone', () => {
        const router = {
            query: {
                search: 'iphone'
            }
        };
        const component = shallow(<Items router={router} />);
        expect(component.getElements()).toMatchSnapshot();
    });

    it('renders item page unchanged on route items/[id]', () => {
        const router = {
            query: {
                id: 'MLA923362627'
            }
        };
        const component = shallow(<Item router={router} />);
        expect(component.getElements()).toMatchSnapshot();
    });

    it('renders error page /error', () => {
        const component = shallow(<Error />);
        expect(component.getElements()).toMatchSnapshot();
    });
});
