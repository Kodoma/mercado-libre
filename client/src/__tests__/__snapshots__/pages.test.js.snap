// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Pages renders error page /error 1`] = `
Array [
  <Layout>
    <ErrorComponent />
  </Layout>,
]
`;

exports[`Snapshot Pages renders homepage unchanged 1`] = `
Array [
  <Layout>
    <React.Fragment />
  </Layout>,
]
`;

exports[`Snapshot Pages renders item page unchanged on route items/[id] 1`] = `
Array [
  <Layout>
    <ItemProduct />
  </Layout>,
]
`;

exports[`Snapshot Pages renders items list page unchanged on route items/search=iphone 1`] = `
Array [
  <Layout>
    <ItemsList />
  </Layout>,
]
`;
